### PR TITLE
Minimum Functional Battery_Test_Jig

### PR DESCRIPTION
**Pull request by ThePakkrat**
Can now set charge pins and sense pins for all battery banks, and there is a basic GUI for testing purposes.

Repository cleaned up.

Readme updated.

One of four chargers (Devpath 1.2) for sure works.

Voltage data looks correct. Temperature data looks incorrect - need a scale factor function to get correct data.

### CODE REVIEW

File: main/libb6/binding.cpp [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+
+
+/*
+Neat Python Tip:
+ *def example(abc: str) -> float: #specifies return type float, abc is a string, print out that arg type is string
+ 	print(abc)
+	return 1.5
+ */
+
```
Comment (ryanpdx) [inline main/libb6/binding.cpp:147]: Remove

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    "Programming Language :: Python :: 3.11",
+    "Topic :: Software Development :: Embedded Systems",
+]
+dependencies = [
+	 "pybind11",
+]
+dynamic = ["version"]
+
+[tool.setuptools.dynamic]
+version = {attr = "oresat_cfc.__version__"}
```
Comment (ryanpdx) [inline pyproject.toml:27]: Remove

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+# R901:     Too complex
+# R0913:    Too many arguments
+# R0914:    Too many local variables
+ignore = "C0103,E203,W0613,R0902,R901,R0913,R0914"
+max_line_length = 100
+
+[[tool.mypy.overrides]]
+module = "canopen,olaf,tifffile,spidev"
+ignore_missing_imports = true
+
```
Comment (ryanpdx) [inline pyproject.toml:49]: Remove

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+pybind11
+black
+isort
+pylama
```
Comment (ryanpdx) [inline requirements.txt:4]: add LabJackPython

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    "Operating System :: POSIX :: Linux",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Topic :: Software Development :: Embedded Systems",
+]
+dependencies = [
+	 "pybind11",
```
Comment (ryanpdx) [inline pyproject.toml:22]: add LabJackPython

File: main/libb6/Device.cc [hunk 6]
```diff
@@ -67,7 +189,8 @@ namespace b6 {
       }
       libusb_close(m_dev);
     }
-    libusb_exit(m_libusbCtx);
+    //libusb_exit(m_libusbCtx); //Without this line, the program works but with memory leaks.
+                                //With this line, the program seg faults.
```
[the remark this answers is not in the document]
Comment (ThePakkrat) [reply main/libb6/Device.cc:193]: No, program no longer seg faults. Works with or without the line.

**Comment by ThePakkrat** [comment]
> Code wise it looks good, mostly just need some clean up.
> 
> Please avoid touching non-sw things on sw changes.
> 
>     * Let's not change the kicad paths, please change `hardware-specs` back to `batter-test-board`.
> 
>     * Make a `docs` dir and move labjack-pin-map.pdf into it
> 
>     * Rename `main/` to `battery_tester/`
> 
>     * Run `black .` and `isort .` to auto-format the Python code
> 
>     * Restore the BOM spreadsheet
> 
>     * Restore `battery-test-board.kicad_prl` and `battery-test-board.kicad_sch-bak`

File: README.md [hunk 7]
```diff
@@ -5,21 +5,27 @@ Battery testing software and circuitry for testing cells and batteries to fly on
...
 ### Setup
-
+- Clone oresat-battery-testing.
 - Install dependencies: `$ sudo apt install libusb-1.0-0-dev pkg-config python3-dev`
-- Initialize libb6 submodule: `$ git submodule update --init`
 - Build Python extension for libb6: `$ make -C libraries`
 
 ### Udev Setup
-Copy the .rules file (from the Documentation folder) into the folder on your computer which contains rules files for udev.
+Copy the .rules files (from the Battery Tester folder) into the folder on your computer which contains rules files for udev.
```
[the remark this answers is not in the document]
Comment (ryanpdx) [reply README.md:16]: We can make a separate task for this.

File: main/libb6/Device.cc (hunk 6 above)
[the remark this answers is not in the document]
Comment (ryanpdx) [reply main/libb6/Device.cc]: Yay!

File: 99-b6mini.rules [hunk 8]
```diff
@@ -0,0 +1,10 @@
+#This is the udev rule for the chargers:
+
+SUBSYSTEM=="usb", ATTRS{idVendor}=="0000", ATTRS{idProduct}=="0001", MODE:="666", GROUP="plugdev"
+KERNEL=="hidraw*", ATTRS{idVendor}=="0000", ATTRS{idProduct}=="0001", MODE:="666", GROUP="plugdev"
+#All devpaths except for 1.2 do not work. This may not be a problem with the devpath itself.
+SUBSYSTEM=="tty",   DEVPATH=="*2.1.4*". SYMLINK+="ttyCHARGERA"
+SUBSYSTEM=="tty",   DEVPATH=="*1.2*". SYMLINK+="ttyCHARGERB"
+SUBSYSTEM=="tty",   DEVPATH=="*2.3*". SYMLINK+="ttyCHARGERC"
+SUBSYSTEM=="tty",   DEVPATH=="*1.4*". SYMLINK+="ttyCHARGERD"
+
```
[the remark this answers is not in the document]
Comment (ThePakkrat) [reply 99-b6mini.rules:10]: I changed the comment in this document. Did you want me to delete those actual four lines?

**Review by ryanpdx — APPROVED** [review]
Looks good!

Only nit-picking at this point

**Pull request merged**